src/App.js

```diff
@@ -1,14 +1,142 @@
+      watchListSection: false
+    }
+    this.receiverSearch = this.receiverSearch.bind(this);
+    this.receiverWatchList = this.receiverWatchList.bind(this);
+    this.receiverDisplayWatchlist = this.receiverDisplayWatchlist.bind(this);
+    this.handleScroll = this.handleScroll.bind(this);
+  }
+
+  handleScroll() {
+    const windowHeight = "innerHeight" in window ? window.innerHeight : document.documentElement.offsetHeight;
+    const body = document.body;
+    const html = document.documentElement;
+    const docHeight = Math.max(body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight);
+    const windowBottom = windowHeight + window.pageYOffset;
+    const backToTop = body.querySelector("#back-to-top");
+
+    const loading = document.querySelector('#search-results__loading');
+    const OMBbAPIKey = "aba7af8e";
+    const query = 'http://www.omdbapi.com/?s=' + this.state.searchTerm + '&apikey=' + OMBbAPIKey + '&page=' + this.state.page;
+
+    // Back to top button
+    if (body.scrollTop > 300 || html.scrollTop > 300) {
+      backToTop.classList.add('visible');
+    } else {
+      backToTop.classList.remove('visible');
+    }
+
+    // Infinite scroll
+    if (windowBottom >= docHeight) {
+      fetch(query)
+        .then(response => {
+          loading.textContent = "Loading...";
+          loading.classList.add('show');
+          return response.json();
+        }
+        )
+        .then(movies => {
+          if (movies.Search) {
+            const self = this;
+            setTimeout(function () {
+              self.setState({
+                search: [...self.state.search, ...movies.Search],
+                page: self.state.page += 1
+              });
+              loading.classList.remove('show');
+            }, 500)
+          } else {
+            loading.textContent = "No more results";
+            loading.classList.add('show');
+          }
+        })
+        .catch(error => {
+          console.log(error);
+          loading.textContent = "There's been a problem fetching the results. Try again later..";
+          loading.classList.add('show');
+        });
+
+    }
+  }
+
+  componentDidMount() {
+    window.addEventListener("scroll", this.handleScroll);
+  }
+
+  receiverSearch(movies, searchTerm) {
+    this.setState({
+      search: movies,
+      searchTerm: searchTerm,
+      watchListSection: false
+    })
+  }
+
+  receiverDisplayWatchlist() {
+    this.setState({
+      watchListSection: true
+    })
+  }
+
+  receiverWatchList(articleTitle, action) {
+    if (action === "addToWatchList") {
+      this.state.search.forEach(articleObj => {
+        return Object.keys(articleObj).find(articleKey => {
+          if (articleObj[articleKey] === articleTitle) {
+            this.setState({
+              watchList: [...this.state.watchList, articleObj]
+            })
+
+            localStorage.getItem("watchList")
```

Comment:
Since we are using localStorage multiple times here, I would suggest using variables to store value from localStorage, update it depending on value found and write updated value back. It will avoid excess reads and make it easier to follow. For example

```js
const watchList = localStorage.getItem("watchList");

const updatedWatchList = watchList ? [articleObj, [...JSON.parse(watchList)] : [...this.state.watchList, articleObj];
 
localStorage.setItem("watchList", JSON.stringify(updatedWatchList));
```